lohsuan/OS_team_pj
# 📚 HW1 Project 1

## Description
This PR is trying to add the chap2 team programming hw for hw1.

Please refer to the README file in the project folder for more details.

## How to verify?
Assume you will run these commands on https://osm.xcc.tw

(for both part1 and part2)
1. compile code
   ```shell
   make
   ```
2. insert the compiled module
   ```shell
   sudo insmod apple.ko [or banana.ko]
   ```
3. remove the module
   ```shell
   sudo rmmod apple [or banana]
   ```
4. Check the kernel log for expected messages
   ```shell
   sudo dmesg
   ```

## Screenshots
- Part1
![](https://i.imgur.com/RlrT7dS.png)

- Part2
![](https://i.imgur.com/hZM9sXU.png)
